src/wallet/api/wallet2_api.h

```diff
@@ -373,6 +373,10 @@ struct WalletListener
  */
 struct Wallet
 {
+    enum Device {
+        Device_Software = 0,
+        Device_Ledger = 1
+    };
```

Comment:
This (well, similar) is already defined in wallet2.h. Is this really needed ?

Reply:
I can't find a similar enum in wallet2.h - which one do you mean? we need it exposed in the api and that's why it needs to be here (even if mirrored/mapped from wallet2.h).

Reply:
Sorry, in device.hpp. It can't be included ?

Reply:
i still don't see it (in master). but it can't be included either way because the api hpp may not pull in internal dependencies - it's standalone (with only c++ dependencies) and needs to map internals to externals to make sure it stays independent and somewhat stable.

Reply:
It's not in master, only in your patch.

Reply:
sorry, yes - this is on purpose. wallet2_api.h is the API and maps the necessary internals independent of them.

Reply:
like https://github.com/monero-project/monero/blob/fad88e18a9795b377676ad6f63c1772868e0d761/src/wallet/api/wallet2_api.h#L44-L48